### ISSUE TITLE
Issue with Training the Victim Model in Attack 2

### ISSUE THREAD

**Issue by Alexzula** [KunlinChoi/POIPrivacy#1]
Dear Team,

Thank you for creating this impressive work. I'm grateful for the opportunity to test the models.

I encountered an issue while running Attack 2. The first command, python build_graph.py, for building the model executed successfully.

However, the second command, python train_target.py, for training the victim model seems to get stuck, as shown in the screenshot below.
![image](https://github.com/user-attachments/assets/d60cf68e-c284-4d97-8785-d3195c08dc80)

Please also note that I successfully ran the first attack model, and the screenshot of the results is attached below for your reference
![Attacker1](https://github.com/user-attachments/assets/20b58433-3d08-4d7c-8ce3-1928bd21693d)

**Comment by KunlinChoi** [comment]
Thank you for your question! I guess it looks stuck because the logs are printed infrequently, only once per epoch. I’ve modified the code to print every 5 training batches instead. Could you check if this resolves the issue?

**Comment by Alexzula** [comment]
Hi KunlinChoi,

Thank you for your prompt response.

Unfortunately, the fix did not resolve the issue. However, I noticed that the training process reached the 4th epoch before getting stuck again.

I will proceed to test the remaining attacks and provide feedback while we work toward a permanent solution for the Attack 2.

Please find the attached file showing the results I obtained before it got stuck at epoch 4.

[POIPrivacy-main - Attack2.txt](https://github.com/user-attachments/files/17408721/POIPrivacy-main.-.Attack2.txt)

Once again, thanks for prompt response.

**Comment by KunlinChoi** [comment]
Can you provide more information on this issue? We have tested it on different devices in our environment, and it seems to be working on our end.

**Comment by Alexzula** [comment]
Thank you, Professor, for your feedback.
I was able to resolve the issue where the Attack 2 simulation was getting stuck.
However, I noticed that it takes a significant amount of time to complete all the EPOCHs. I'm not sure if this delay is due to the configuration of my laptop. For reference, I am using a Dell XPS 13 7390 with the following specifications:

Processor: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz, 2304 MHz, 4 Cores, 8 Logical Processors
Installed Physical Memory (RAM): 16.0 GB
Could this be a factor in the slow performance?
